tds.go

```diff
@@ -273,6 +273,10 @@ func readPreloginOption(buffer []byte, offset int) (*preloginOption, error) {
 	rec_type := buffer[offset]
 	if rec_type == preloginTERMINATOR {
 		return &preloginOption{token: rec_type}, nil
+	} else if rec_type == preloginTHREADID {
```

Comment:
>rec_type == preloginTHREADID { [](http://example.com/codeflow?start=11&length=30)

does this check need to be protected by the TDS version check too?

Reply:
As per https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-tds/60f56408-0188-4cd5-8b90-25c6f2423868 there seems to be no constraint on the TDS-Version. 

As I understand the documentation, the parameter is used for the client to communicate its thread-id for debugging-purposes. If sent by the server, it should be empty in any case.